libs/features/georoutingv2/georoutingv2.js

```diff
@@ -280,12 +282,19 @@ export default async function loadGeoRouting(
   loadBlock = loadBlockFunc;
   loadStyle = loadStyleFunc;
 
-  const resp = await fetch(`${config.contentRoot ?? ''}/georoutingv2.json`);
+  let resp = await fetch(`${config.contentRoot ?? ''}/georoutingv2.json`);
   if (!resp.ok) {
-    // eslint-disable-next-line import/no-cycle
-    const { default: loadGeoRoutingOld } = await import('../georouting/georouting.js');
-    loadGeoRoutingOld(config, createTag, getMetadata);
-    return;
+    resp = await fetch(`${config.contentRoot ?? ''}/georouting.json`);
+    if (!resp.ok) {
+      resp = await fetch(`${getFederatedContentRoot()}/federal/georouting/georoutingv2.json`);
+      if (!resp.ok) return;
+    } else {
+      const json = await resp.json();
+      // eslint-disable-next-line import/no-cycle
+      const { default: loadGeoRoutingOld } = await import('../georouting/georouting.js');
+      loadGeoRoutingOld(config, createTag, getMetadata, json);
+      return;
+    }
```

Comment:
Should the failover order be `georoutingv2 -> geouroutingv1 -> georoutingv2Federated` or `georoutingv2 -> georoutingv2Federated -> georoutingv1`? It's not obvious to me which one it should be from reading the ticket.

If it were to be the latter you could do something like this:
```javascript
let json;
try {
  json = await loadGeoRoutingJson(...args); // loads either non-federal, federal, or throws error
} catch (e) {
  // maybe some logging in case resp.json() errors out
  loadGeoRoutingOld(...args);
  return;
}
```
This would put the fetching of individual urls back in the hands of the respective functions (which means you won't have to change georouting.js at all), it would clean up the above logic, and it would handle `await resp.json()` potentially erroring out without being handled.

Overall I think a structure like this would make the code both safer (as in able to fail gracefully should something unexpected happen in a network request, say in a slow network, that causes either the fetch or the `resp.json()` to fail) and easier to read.

Reply:
The failover would be this - georoutingv2 -> geouroutingv1 -> georoutingv2Federated. This is why we need to check both versions first, and if neither is found, then fetch from the federal repository.